File: net-api/net-api/Models/Spell.cs
using Newtonsoft.Json;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace net_api.Models
{
    public class Spell
    {
        public Guid Id { get; set; }

        [Required]
        [StringLength(30, MinimumLength = 3)]
        public string Name { get; set; }

        [Column("Content", TypeName = "JSONB")]
        public string ContentJson { get; set; }

        [NotMapped]
        public Object Content
        {
            get
            {
                if (ContentJson == null)
                {
                    return null;
                }

                return JsonConvert.DeserializeObject(ContentJson);
            }

            set
            {
                ContentJson = JsonConvert.SerializeObject(value);
            }
        }

        public string ImageId { get; set; }

        [Required]
        public bool PlayerVisible { get; set; }

        [Required]
        public Guid CampaignId { get; set; }

        [JsonIgnore]
        public Campaign Campaign { get; set; }

        public string UserId { get; set; }
        public User User { get; set; }

        [Column("Tags", TypeName = "varchar[]")]
        public string[] Tags { get; set; }
    }
}

File: data-tools/inout-lambda/spell.go
package main

import "upper.io/db.v3/postgresql"

type Spell struct {
	Name          string                 `db:"Name" json:"name"`
	Content       postgresql.JSONB       `db:"Content" json:"content"`
	PlayerVisible bool                   `db:"PlayerVisible" json:"playerVisible"`
	Tags          postgresql.StringArray `db:"Tags" json:"tags"`
	ImageID       *string                `db:"ImageId" json:"-"`
}
